src/window.rs

```diff
@@ -943,12 +944,26 @@ impl Window {
         self.window.request_user_attention(request_type)
     }
 
-    /// Returns the current window theme.
+    /// Sets the current window theme. Use `None` to fallback to system default.
     ///
     /// ## Platform-specific
     ///
+    /// - **Wayland:** You can also use `WINIT_WAYLAND_CSD_THEME` env variable to set the theme.
+    /// Possible values for env variable are: "dark" and light". When unspecified, a theme is automatically selected.
+    /// -**x11:** Sets `_GTK_THEME_VARIANT` hint to `dark` or `light`.
+    /// -**x11 / macOS**: `None` will default to  [`Theme::Dark`].
```

Comment:
I would expect that `None` on macOS just makes it use the default behaviour (which would be to change theme automatically when the system does)?

Reply:
Should probably also note that on macOS this is an application global, not actually something to set on the window

Reply:
> Perhaps it would make sense to add `Theme::Auto`, and then just use `Theme` instead of `Option<Theme>`?
> Though I guess that depends on what kind of information one desires from `Window::theme` (is it the currently active theme, or the setting)?

I'd imagine the current active theme.

> I would expect that `None` on macOS just makes it use the default behaviour (which would be to change theme automatically when the system does)?

Yes, but it wasn't clear to me how to do, which you explained below,  thanks.

> Should probably also note that on macOS this is an application global, not actually something to set on the window

Should I remove this method from `Window` for macOS and add it to `EventLoopExtMacOS` instead?

Reply:
> Should I remove this method from `Window` for macOS and add it to `EventLoopExtMacOS` instead?

No, I think there is value in having a unified API since most users are only going to be having one window anyways. But I think it'd be nice to have documented!

Reply:
Added the notes. Let me know if you want to me to add `Theme::Auto`, I think it can be acheived for macOS, Windows and Wayland. (not sure about x11, couldn't find anything related to the matter).

Reply:
Let's just keep using `Option<Theme>`, since it's probably strictly more correct as the return type for `Window::theme`